### PR TITLE
Bugfix: Added the ability to refer to 'chars' and '()'  in type specs.

### PR DESCRIPTION
**Pull request by danabr**
Also made sure these are accepted as literals.

### CODE REVIEW

File: test_files/values.alp [hunk 1]
```diff
@@ -15,6 +18,20 @@ let test_atom = :nope
 let test_bool = true
 let test_chars = c"x"
 let test_tuple = (100, "100")
+let test_unit = ()
+
+let wrapped_int = AnInt 41
```
Comment (danabr) [inline test_files/values.alp]: @lepoetemaudit: I tried writing
````
````
but then I got `{error,non_literal_value, ...`.  Perhaps we should allow arbitrary names in here, and resolve the references in alpaca_ast_gen. What do you think?

Comment (j14159) [inline test_files/values.alp]: @danabr @lepoetemaudit I think maybe we just need to allow for constructors as literal values?  The typer should catch any further issues.

Comment (danabr) [inline test_files/values.alp]: Or rather allow symbols as literal values (literal is not as fitting a name then, perhaps), and then have the typer or ast_gen stage complain if the symbol doesn't refer to a zero arity function.

Constructors as literals are already supported. as the test witnesses.

Comment (j14159) [inline test_files/values.alp]: Quite right, apologies!  I would blame insufficient coffee this morning but really I should have just looked closer :)

@danabr if you want this merged prior to discussion/changes between you and @lepoetemaudit  or prefer I hold off just let me know.  I'm fine either way.

Comment (danabr) [inline test_files/values.alp]: I'm fine with merging this. The purpose of this PR was to add `chars`and `()` as basic types. I just happened to notice the described behavior when implementing this and wondered if it was intentional.

Comment (j14159) [inline test_files/values.alp]: OK, will merge for now.  Thanks!

**Pull request merged**